tests/unit/test_kwargs/test_multipart_data.py

```diff
@@ -597,3 +618,100 @@ def test_invalid_multipart_raises_client_error() -> None:
+    """Test that empty strings are preserved in multipart forms and not converted to None.
+
+    This test addresses GitHub issue #4204 where empty strings in multipart/form-data
+    requests were being converted to None instead of being preserved as empty strings.
+    The test focuses on core multipart parsing behavior, not serialization library specifics.
+    """
+
+    @post("/test-form")
+    async def form_handler(request: Request) -> dict[str, Any]:
+        """Handler that directly tests form parsing without any serialization library."""
+        data = await request.form()
+        return {
+            "value": data.get("value"),
+            "value_type": type(data.get("value")).__name__,
+            "value_is_none": data.get("value") is None,
+        }
+
+    with create_test_client([form_handler]) as client:
+        # Test URL-encoded form (baseline behavior)
+        response_url_encoded = client.post("/test-form", data={"value": ""})
+        assert response_url_encoded.status_code == HTTP_201_CREATED
+        url_result = response_url_encoded.json()
+
+        # Test multipart form (should behave identically to URL-encoded)
+        response_multipart = client.post("/test-form", data={"value": ""}, files={"dummy": ""})
+        assert response_multipart.status_code == HTTP_201_CREATED
+        multipart_result = response_multipart.json()
+
+        # Both should preserve empty strings, not convert to None
+        assert url_result["value"] == ""
+        assert multipart_result["value"] == ""
+        assert url_result["value_type"] == "str"
+        assert multipart_result["value_type"] == "str"
+        assert url_result["value_is_none"] is False
+        assert multipart_result["value_is_none"] is False
+
+        # Results should be identical (consistency between form types)
+        assert url_result == multipart_result
+
+
+def test_empty_strings_consistency_between_encodings() -> None:
+    """Test that empty strings behave consistently between URL-encoded and multipart forms."""
+
+    @post("/consistency-test")
+    async def consistency_handler(request: Request) -> dict[str, Any]:
+        data = await request.form()
+        return {"value": data.get("value"), "value_type": type(data.get("value")).__name__}
+
+    with create_test_client([consistency_handler]) as client:
+        # Test URL-encoded form
+        response_url = client.post("/consistency-test", data={"value": ""})
+        url_result = response_url.json()
+
+        # Test multipart form
+        response_multipart = client.post("/consistency-test", data={"value": ""}, files={"dummy": ""})
+        multipart_result = response_multipart.json()
+
+        # Both should return empty string, not None
+        assert url_result["value"] == ""
+        assert multipart_result["value"] == ""
+        assert url_result["value_type"] == "str"
+        assert multipart_result["value_type"] == "str"
```

Comment:
This is already covered by the other tests. You should try to keep these units tests minimal and focused; They ideally should only test *one* thing. What this test wants to show is that the multipart / url encoded form data decodes the same, so there should only be one assert needed for this.

This makes the tests easier to follow, since their intention is more clear.